src/main/java/net/openhft/chronicle/queue/impl/single/ThreadLocalAppender.java

```diff
@@ -1,16 +1,39 @@
 package net.openhft.chronicle.queue.impl.single;
 
+import net.openhft.chronicle.queue.ChronicleQueue;
 import net.openhft.chronicle.queue.ExcerptAppender;
 
-public class ThreadLocalAppender {
+/**
+ * Internal implementation class that enables the use of a thread local appender. Please ensure that you carefully read
+ * the javadoc and understand the lifecycle of this thread-local appender before use.
+ */
+public final class ThreadLocalAppender {
+
+    private ThreadLocalAppender() {
+        // Intentional no-op
+    }
 
     /**
-     * A thread local appender for the SingleChronicleQueue, this will give you the same thread local instance per thread.
+     * Returns a ExcerptAppender for the given ChronicleQueue that is local to the current Thread.
+     * <p>
+     * An Appender can be used to store new excerpts sequentially to the queue.
+     * <p>
+     * <b>
+     * An Appender is <em>NOT thread-safe</em> and, in addition to that, confined to be used <em>by the creating thread only.</em>.
+     * Sharing an Appender across threads is unsafe and will inevitably lead to errors and unspecified behaviour.
+     * </b>
+     * <p>
+     * This method returns a {@link ThreadLocal} appender, so does not produce any garbage, hence it's safe to simply call
+     * this method every time an appender is needed.
      *
-     * @param queue the SingleChronicleQueue
-     * @return a thread local appender
+     * @return Returns a ExcerptAppender for this ChronicleQueue that is local to the current Thread
+     * @throws IllegalArgumentException if the queue it is passed is not an instance of {@link SingleChronicleQueue}
      */
-    public static ExcerptAppender acquireThreadLocalAppender(SingleChronicleQueue queue) {
-        return queue.acquireThreadLocalAppender(queue);
+    public static ExcerptAppender acquireThreadLocalAppender(ChronicleQueue queue) {
```

Comment:
@JerryShea, @RobAustin makes a good point - there is a case to leave this accepting `SingleChronicleQueue` because this class is tightly bound to the implementation itself, not something that is available on the interface. By accepting `ChronicleQueue` here we may encourage others to pass any queue into this method. Thoughts?

Reply:
We could create an instance of `ThreadLocalAppender` for a queue and have *it* manage the TL (and implement Closeable etc.) and then (I haven't checked this in detail) maybe we could remove the TL from `SingleChronicleQueue` and it would work for all types of queues.

Reply:
I like the idea @JerryShea, but it would still pose issues for method writers (when do we create and close the `ThreadLocalAppender` instance), the fundamental issue is they don't have a lifecycle and are themselves "thread safe".

I believe our goal here should be making this API as obscure and hidden as possible, while reserving the right to rip it out as soon as we fix the stragglers that require it (channels and method writers from what I can see). If we made those things single threaded and added a lifecycle to them we can put this whole mess behind us and let the API consumer do their resource management and thread-localisation if that's what they desire.